### PR TITLE
Remove char_width member from cudf::string_view class

### PR DESCRIPTION
**Pull request by davidwendt**
This PR removes the `_char_width` member from the libcudf `cudf::string_view` class. This member was being used to record the character width of the bytes in the string but only if all the characters have the same width. This occurs when the string only contains ASCII encoded data which is all single-byte UTF-8 characters. The same optimization can be inferred when the existing `_length` and `_bytes` are equal. 
This change reduces the memory footprint of this class from 24 bytes to 16 bytes and therefore matches the size of `thrust::pair<char*,size_type>`. Using this class in a vector would thereby reduce the memory requirements for that vector by 1/3.

### CODE REVIEW

File: cpp/include/cudf/strings/string_view.hpp [hunk 1]
```diff
@@ -314,7 +307,7 @@ class string_view {
   /**
    * @brief Default constructor represents an empty string.
    */
-  CUDA_HOST_DEVICE_CALLABLE string_view() : _data(""), _bytes(0), _length(0), _char_width(0) {}
+  CUDA_HOST_DEVICE_CALLABLE string_view() : _data(""), _bytes(0), _length(0) {}
```
Comment (vuule) [inline cpp/include/cudf/strings/string_view.hpp:310]: What does `_data("")` do here? `""` is a temporary, and `_data` will end up pointing to it, which should be invalid as soon as we exit the constructor. I must be missing something.

**Review by vuule — COMMENTED** [review]
Looks good, just got a question about the default constructor.

File: cpp/include/cudf/strings/string_view.hpp (hunk 1 above)
Comment (davidwendt) [reply cpp/include/cudf/strings/string_view.hpp:310]: The `""` is a constant. It is intended to ensure the default ctor creates an instance representing an empty string and not a nullptr. This is just a convenience for use in some `thrust` and other utility functions.

**Review by vuule — APPROVED** [review]

**Review by rgsl888prabhu — APPROVED** [review]

**Comment by hyperbolic2346** [comment]
We don't support a multi-byte, fixed-width string type like unicode?

**Comment by davidwendt** [comment]
> We don't support a multi-byte, fixed-width string type like unicode?

That's right, libcudf only supports UTF-8.

**Review by hyperbolic2346 — APPROVED** [review]

**Comment by davidwendt** [comment]
@gpucibot merge

**Pull request merged**